### PR TITLE
Add paperless-ng

### PR DESCRIPTION
**Pull request by phiranf**
**What this PR does / why we need it**:

This PR adds the paperless-ng project. It's very useful to digitize, manage and search all your documents.
[Take a look at the projects github repo](https://github.com/jonaswinkler/paperless-ng)

**Which issue (if any) this PR fixes**:

\--

**Any other useful info**:
The way the volumes are mapped is how I personally use them. This possibly can be improved.

### CODE REVIEW

File: docs/applications/paperless_ng.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+Set `paperless_enabled: true` in your `inventories/<your_inventory>/nas.yml` file.
+
+The paperless-ng web interface can be found at http://ansible_nas_host_or_ip:16922.
+
+### Create the superuser
+
+To be able to login, you will need a super user. To create it, execute the following command:
+
+```bash
+docker-compose run --rm ng_paperless_uiserver createsuperuser
```
Comment (davestephens) [inline docs/applications/paperless_ng.md:20]: I'm not clear how this would work?

File: group_vars/all.yml [hunk 2]
```diff
@@ -105,6 +105,9 @@ comics_root: "{{ samba_shares_root }}/comics"
 # Where photos are stored
 photos_root: "{{ samba_shares_root }}/photos"
 
+# Where paperless_ng consumes documents
+consume_documents: "{{ samba_shares_root }}/consume_documents"
```
Comment (davestephens) [inline group_vars/all.yml:109]: This should really be `documents_root` and "consume" could be a subdirectory of that. The various roots defined are types of media, ie music, movies, comics etc.

File: group_vars/all.yml [hunk 3]
```diff
@@ -198,6 +201,14 @@ samba_shares:
     browseable: yes
     path: "{{ comics_root }}"
 
+  - name: consume
```
Comment (davestephens) [inline group_vars/all.yml:204]: Should be `documents`

File: roles/paperless_ng/defaults/main.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+---
+paperless_ng_enabled: false
+
+# directories
+paperless_data_directory: "{{ docker_home }}/ng_paperless"
```
Comment (davestephens) [inline roles/paperless_ng/defaults/main.yml:5]: Var prefixes should be consistent - typical standard is that they match the role name as you have done with `paperless_ng_enabled`.

Directory names should match the role name too, `ng_paperless` feels backwards.

File: roles/paperless_ng/defaults/main.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+paperless_port: "16922"
+paperless_hostname: "ng_paperless"
+
+# postgres
+paperless_postgres_db: "paperless"
+paperless_postgres_user: "paperless"
+paperless_postgres_password: "paperless"
+
+#container names
+paperless_container_network_name: "ng_paperless_network"
```
Comment (davestephens) [inline roles/paperless_ng/defaults/main.yml:18]: Naming starting with `ng_` feels backwards?

**Comment by davestephens** [comment]
Thanks for the contribution, not something I've heard of before and looks very interesting! 

I've put a few comments in the code that'll need looking at. Also;

- please add it to the readme.md
- please add memory limits on the containers
- you have a minor linting issue to fix

Thanks! 👍

File: docs/applications/paperless_ng.md (hunk 1 above)
Comment (phiranf) [reply docs/applications/paperless_ng.md]: It didn't, I just copy and pasted it from the documentation. Fixed now.

**Comment by phiranf** [comment]
I implemented your comments but still need to test it. I currently don't get the provided testing scripts to run on my mac or windows machine though. I'll post an update here when I verified everything is indeed working as it should :)

**Comment by davestephens** [comment]
Looks good! Let me know how you get on with testing.

**Comment by phiranf** [comment]
I've just tested it on my laptop with a fresh Ubuntu 20.04 installation and everything is working now. If you like we can merge it now.

**Pull request merged**